cognition_objects/message.py

```diff
@@ -189,3 +195,133 @@ def delete(project_id: str, message_id: str, with_commit: bool = True) -> None:
         CognitionMessage.id == message_id,
     ).delete()
     general.flush_or_commit(with_commit)
+
+
+def get_messages_per_conversation_query(project_id: str) -> str:
+    project_id = prevent_sql_injection(project_id, isinstance(project_id, str))
+    return f"""
+    SELECT 
+        conversation_id,
+        COUNT(*) messages,
+        c."header"
+    FROM cognition.message as m
+        inner join cognition.conversation c on c.id = m.conversation_id and c.project_id = m.project_id 
+    WHERE m.project_id = '{project_id}'
+    GROUP BY conversation_id, c."header" 
+    ORDER BY conversation_id, c."header"
+    """
+
+
+def get_response_time_messages_query(project_id: str) -> str:
+    project_id = prevent_sql_injection(project_id, isinstance(project_id, str))
+    return f"""
+    select round(sum(x) * 1000 / 500) * 500 / 1000 AS time_seconds, COUNT(*)
+    from (
+        select m.id, sum(pl.time_elapsed)
+    from cognition.message m
+        inner join cognition.conversation c on c.id = m.conversation_id and c.project_id = m.project_id
+        inner join cognition.pipeline_logs pl ON m.id = pl.message_id 
+    WHERE m.project_id = '{project_id}'
+    group by m.id
+    ) x
+    group by time_seconds
+    """
+
+
+def get_conversations_messages_count_query(project_id: str) -> str:
+    project_id = prevent_sql_injection(project_id, isinstance(project_id, str))
+    return f"""
+    SELECT 
+        COUNT(*) num_conversations,
+        num_messages,
+        COUNT(*) / (SELECT COUNT(*) FROM cognition.conversation WHERE project_id = '{project_id}')::FLOAT * 100 percentage
+    FROM (
+        SELECT 
+            conversation_id,
+            COUNT(*) num_messages
+        FROM cognition.message as m
+            inner join cognition.conversation c on c.id = m.conversation_id and c.project_id = m.project_id 
+        WHERE m.project_id = '{project_id}'
+        GROUP BY conversation_id
+    ) x
+    GROUP BY num_messages
+    ORDER BY num_messages
+    """
```

Comment:
we usually dont use selects directly in the select part as it's unreadable.
As the count only has one record the same can be achieved via a cross apply.
Further the join to conversation for the subselect isn't necessary
```suggestion
    return f"""
SELECT 
	num_messages,
	num_conversations,
	num_conversations / conv_count.c * 100 percentage
FROM (
	SELECT COUNT(*) num_conversations, num_messages
	FROM (
	  SELECT conversation_id, COUNT(*) num_messages
	  FROM cognition.message as m
	  WHERE m.project_id = '{project_id}'
	  GROUP BY conversation_id
	) x
	GROUP BY num_messages 
)x,
(SELECT COUNT(*)::FLOAT c FROM cognition.conversation WHERE project_id = '{project_id}') conv_count
ORDER BY 1
    """
```